numpy/ma/core.py

```diff
@@ -6535,9 +6530,7 @@ def compressed(x):
         Equivalent method.
 
     """
-    if not isinstance(x, MaskedArray):
-        x = asanyarray(x)
-    return x.compressed()
+    return asanyarray(x).compressed()
```

Comment:
I'll add another commit for this that fixes a bunch of other methods as well, that were out of scope for the original PR